packages/ios_platform_images/ios/Classes/IosPlatformImagesPlugin.m

```diff
@@ -40,6 +40,99 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
+        // see
+        // https://developer.apple.com/documentation/uikit/uiimagesymbolconfiguration/3810054-configurationwithpalettecolors
+        NSArray *rgbaValuesList = call.arguments[3];
+
+        NSMutableArray *colorArray = [[NSMutableArray alloc] init];
+
+        for (int i = 0; i < [rgbaValuesList count]; i += 4) {
+          UIColor *primaryColor = [UIColor colorWithRed:[rgbaValuesList[i] doubleValue]
+                                                  green:[rgbaValuesList[i + 1] doubleValue]
+                                                   blue:[rgbaValuesList[i + 2] doubleValue]
+                                                  alpha:[rgbaValuesList[i + 3] doubleValue]];
+          [colorArray addObject:primaryColor];
+        }
+
+        UIImageSymbolWeight weight = UIImageSymbolWeightRegular;
+
+        switch ([weightIndex integerValue]) {
+          case 0:
+            weight = UIImageSymbolWeightUltraLight;
+            break;
+          case 1:
+            weight = UIImageSymbolWeightThin;
+            break;
+          case 2:
+            weight = UIImageSymbolWeightLight;
+            break;
+          // 3 is regular
+          case 4:
+            weight = UIImageSymbolWeightMedium;
+            break;
+          case 5:
+            weight = UIImageSymbolWeightSemibold;
+            break;
+          case 6:
+            weight = UIImageSymbolWeightBold;
+            break;
+          case 7:
+            weight = UIImageSymbolWeightHeavy;
+            break;
+          case 8:
+            weight = UIImageSymbolWeightBlack;
+            break;
+          default:
+            weight = UIImageSymbolWeightRegular;
+            break;
+        }
+
+        UIImageSymbolConfiguration *pointSizeConfig =
+            [UIImageSymbolConfiguration configurationWithPointSize:pointSize
+                                                            weight:weight
+                                                             scale:UIImageSymbolScaleDefault];
+
+        UIImage *finalImage;
+
+        if (@available(iOS 15, *)) {
+          NSNumber *preferMulticolor = call.arguments[4];
+          UIImageSymbolConfiguration *colors;
+
+          if ([preferMulticolor boolValue]) {
+            colors = [UIImageSymbolConfiguration configurationPreferringMulticolor];
+          } else {
+            colors = [UIImageSymbolConfiguration configurationWithPaletteColors:colorArray];
+          }
+
+          UIImageSymbolConfiguration *final =
+              [pointSizeConfig configurationByApplyingConfiguration:colors];
+          finalImage = [UIImage systemImageNamed:name withConfiguration:final];
+        } else {
+          UIImage *image = [UIImage systemImageNamed:name withConfiguration:pointSizeConfig];
+          finalImage = [image imageWithTintColor:colorArray.count > 0 ? colorArray[0] : nil];
+        }
+
+        NSData *data = UIImagePNGRepresentation(finalImage);
```

Comment:
Is this documented as allowing `nil` as an argument?

Reply:
It makes the image black for me, but it's not officially documented on https://developer.apple.com/documentation/uikit/uiimage/3327300-withtintcolor, so I changed it to use black directly.